policy_definitions/network_security/policy_definition_qby_allow_subnet_nsg.json

```
@@ -0,0 +1,287 @@
+                          "match": "###"
+                        }
+                      ]
+                    },
+                    {
+                      "anyOf": [
+                        {
+                          "value": "[if(greaterOrEquals(length(split(field('name'), '-')), 6), split(field('name'), '-')[5], 'length of name incorrect')]",
+                          "match": "#"
+                        },
+                        {
+                          "value": "[if(greaterOrEquals(length(split(field('name'), '-')), 6), split(field('name'), '-')[5], 'length of name incorrect')]",
+                          "match": "##"
+                        }
+                      ]
+                    }
+                  ]
+                }
+              }
+            ]
+          },
+          {
+            "allOf": [
+              {
+                "field": "type",
+                "equals": "Microsoft.Network/virtualNetworks"
+              },
+              {
+                "not": {
+                  "allOf": [
+                    {
+                      "count": {
+                        "field": "Microsoft.Network/virtualNetworks/subnets[*]",
+                        "where": {
+                          "not": {
+                            "anyOf": [
+                              {
+                                "field": "Microsoft.Network/virtualNetworks/subnets[*].name",
+                                "in": [
+                                  "GatewaySubnet",
+                                  "AzureFirewallSubnet",
+                                  "RouteServerSubnet",
+                                  "AzureBastionSubnet"
+                                ]
+                              },
+                              {
+                                "AllOf": [
+                                  {
+                                    "field": "Microsoft.Network/virtualNetworks/subnets[*].name",
+                                    "like": "snet-*"
+                                  },
+                                  {
+                                    "value": "[length(split(string(field('Microsoft.Network/virtualNetworks/subnets[*].name')), '-'))]",
+                                    "greaterOrEquals": 7
+                                  },
+                                  {
+                                    "field": "Microsoft.Network/virtualNetworks/subnets[*].networkSecurityGroup.id",
+                                    "exists": true
+                                  },
+                                  {
+                                    "value": "[last(split(string(field('Microsoft.Network/virtualNetworks/subnets[*].networkSecurityGroup.id')),'networkSecurityGroups/'))]",
+                                    "equals": "[concat('nsg-',first(split(last(split(field('id'),'virtualNetworks/')),'/subnets')))]"
```

It shouldn't hurt to split unnecessarily. But to be consistent with the condition in subnet type, we should use the subnet ID instead of vNet
```suggestion
                                    "equals": "[concat('nsg-',first(split(last(split(string(field('Microsoft.Network/virtualNetworks/subnets[*].id')),'virtualNetworks/')),'/subnets')))]"
```

---

did not work